### PR TITLE
#MAN-911 update people controller for performance

### PR DESCRIPTION
**Pull request by cdoyle-temple**

### CODE REVIEW

File: app/controllers/persons_controller.rb [hunk 1]
```diff
@@ -18,84 +14,66 @@ def index
 
   def show
     @buildings = @person.spaces.map { |space| space.building }.uniq
-    @departments = @person.groups.sort.select { |group| group.group_type == "Department" }
+    @departments = @person.groups.select { |group| group.group_type == "Department" }.sort
```
Comment (nomadicoder) [inline app/controllers/persons_controller.rb:17]: This is stylistic - Let's `.sor` on its own line so we ca more easily see the chaining.

File: app/controllers/persons_controller.rb [hunk 2]
```diff
@@ -18,84 +14,66 @@ def index
...
-    filtered_people = [Array(location_people), Array(department_people), Array(special_people)].reject(&:empty?).reduce(:&) || []
+  def get_persons
+    @filtered_persons = Person
+                        .is_specialist(params[:specialists])
+                        .with_specialty(params[:specialty])
+                        .at_location(params[:location])
+                        .in_department(params[:department])
+                        .order(:last_name, :first_name)
+
+    @persons_list = @filtered_persons.page params[:page]
```
Comment (nomadicoder) [inline app/controllers/persons_controller.rb:55]: Enclose `params[:page]` in parenthesis

File: app/controllers/persons_controller.rb [hunk 3]
```diff
@@ -18,84 +14,66 @@ def index
...
+    params[:location]
+  end
 
   private
     def set_person
       @person = Person.friendly.find(params[:id])
-      @department = @person.groups.collect.reject { |g| g.group_type != "Department" }.uniq.sort
+      @department = @person.groups.collect { |g| g.group_type == "Department" }.uniq.sort
       @location = @person.spaces.collect.uniq.sort
     end
```
Comment (nomadicoder) [inline app/controllers/persons_controller.rb:78]: Break out the processing steps into separate lines for department and location as noted above.

File: app/controllers/persons_controller.rb (hunk 1 above)
Comment (cdoyle-temple) [reply app/controllers/persons_controller.rb:17]: ok

File: app/controllers/persons_controller.rb (hunk 2 above)
Comment (cdoyle-temple) [reply app/controllers/persons_controller.rb:55]: ok

File: app/controllers/persons_controller.rb (hunk 3 above)
Comment (cdoyle-temple) [reply app/controllers/persons_controller.rb:78]: Removed lines as redundant code.

**Pull request merged**